src/ImageSharp/Formats/Tiff/Writers/TiffBiColorWriter{TPixel}.cs

```diff
@@ -36,50 +36,38 @@ public TiffBiColorWriter(ImageFrame<TPixel> image, MemoryAllocator memoryAllocat
         /// <inheritdoc/>
         protected override void EncodeStrip(int y, int height, TiffBaseCompressor compressor)
         {
-            int width = this.Image.Width;
+            this.pixelsAsGray ??= this.MemoryAllocator.Allocate<byte>(height * this.Image.Width);
+
+            Span<byte> pixelAsGraySpan = this.pixelsAsGray.Slice(0, height * this.Image.Width);
+
+            Span<TPixel> pixelsBlackWhite = GetStripPixels(this.imageBlackWhite.GetRootFramePixelBuffer(), y, height);
```

Comment:
This is using `GetSingleSpan()`, which might be an issue when the buffer is discontiguous. We should do the conversion line-by-line instead.

In long term, it would be cool to have `PixelOperations<TPixel>` overloads which work with `Buffer2DRegion<T>`s, so we can work with "image slices" more easily.

Reply:
That's weird, I have changed that yesterday with #1662. Did something went wrong here with merging master? Usage of  `GetSingleSpan()` was removed. See: [TiffBiColorWriter, master](https://github.com/SixLabors/ImageSharp/blob/master/src/ImageSharp/Formats/Tiff/Writers/TiffBiColorWriter%7BTPixel%7D.cs)

Reply:
WTH? Something bust have been funky about one of the recent PRs to do this. I've just been pulling master straight in. I'll have to fix it all manually.

Reply:
I cannot really tell how this happened. 

I have added tests for discontiguous buffers: 23093821

Reply:
I'm just about to force push a restored branch which will overwrite your commit.

Reply:
The test commit? Ok that's easy to replace, go ahead.

Reply:
Done

Reply:
> In long term, it would be cool to have `PixelOperations<TPixel>` overloads which work with `Buffer2DRegion<T>`s, so we can work with "image slices" more easily.

+1 for this.
It can be a game changer for jpeg encoding.